R/tfpscanner.R

```diff
@@ -62,698 +70,796 @@ tfpscan <- function(tre,
-			gtr <- .cluster_tree(tu)
-			suppressMessages(
-			  ggplot2::ggsave(gtr,
-			                  filename = glue::glue("{cldir}/clustertree.pdf"),
-			                  height = max(6, floor(length(tu) / 5)),
-			                  width = min(44, max(24, sqrt(length(tu)))),
-			                  limitsize = FALSE)
-			)
-		}
-		# clock figure TODO
-		# tip table
-		utils::write.csv(amd[amd$sequence_name %in% tu, ],
-		                 file = glue::glue("{cldir}/sequences.csv"))
-		# reg summary
-		utils::write.csv(reg_summary,
-		          file = glue::glue("{cldir}/regional_composition.csv"))
-		# lineage summary
-		utils::write.csv(lineage_summary,
-		          file = glue::glue("{cldir}/lineage_composition.csv"))
-		# cocirc lineage summary
-		utils::write.csv(cocirc_summary,
-		          file = glue::glue("{cldir}/cocirculating_lineages.csv"))
-		X
-	}
-
-	# main analysis thread
-	## compute stats for subset of nodes based on size and age
-	nodes <- which((ndesc >= min_descendants) & (ndesc <= max_descendants) & (clade_age >= min_cluster_age_yrs))
-	nodes_blenConstraintSatisfied <- tre$edge[tre$edge.length > min_blen, 2]
-	nodes <- intersect(nodes, nodes_blenConstraintSatisfied)
-	report_nodes <- nodes[seq(1, length(nodes), by = report_freq)] #progress reporting
-	node_ancestors <- do.call(c, lapply(nodes, function(u) ancestors[[u]]))
-	if (ncpu > 1) {
-		message("Initiating MPI cluster")
-		mpiclust <- doMPI::startMPIcluster(count = ncpu)
-		doMPI::registerDoMPI(mpiclust)
-		foreach::foreach(u = nodes,
-		        .combine = rbind,
-		        .packages = c("lubridate", "glue", "mgcv", "ggplot2", "ggtree", "phangorn"),
-		        .export = c("output_dir"),
-		        .errorhandling = "remove",
-		        .verbose = TRUE
-		) %dopar% {
-			tryCatch(.process.node(u),
-			         error = function(e) {
-			           saveRDS(e, file = glue::glue("{u}-err.rds"))
-			           return(e)
-			           }
-			         )
-		} -> Y # nolint
-	} else {
-		Y <- c()
-		for (u in nodes) {
-			X <- .process.node(u)
-			Y <- rbind(Y, X)
-		}
-	}
-	if (!all(nodes %in% Y$node_number)) {
-		stop("Statistics not computed for all nodes. Possible that memory exceeded with ncpu > 1. Try with ncpu = 1.")
-	}
-	Y <- Y[order(Y$logistic_growth_rate, decreasing = TRUE), ]
-
-	ofn1 <- glue::glue("{output_dir}/scanner-{max_date}.rds")
-	ofn3 <- glue::glue("{output_dir}/scanner-env-{max_date}.rds")
-	saveRDS(Y, file = ofn1)
-	message("saving image ... ")
-	# save internal variables and functions
-	e0 <- environment()
-	saveRDS(e0, file = ofn3)
-	message(glue::glue("Data written to {ofn1} and {ofn3}. Returning data frame invisibly."))
-
-	if (ncpu  > 1) {
-		doMPI::closeCluster(mpiclust)
-		Rmpi::mpi.finalize()
-	}
-	invisible(Y)
+                    root_on_tip_sample_time = 2020,
+                    detailed_output = FALSE,
+                    compute_gam = TRUE,
+                    compute_cluster_muts = TRUE) {
+  message(paste("Starting scan", Sys.time()), "\n")
```

Comment:
styler was used on jr/tfpscanner:master in 5693bfe . It will be easier to see any functional changes by diffing against that commit.

Reply:
I've compared against this commit and it looks good